test/unit/ngsiv2/HTTP_reveice_measures-test.js

```diff
@@ -0,0 +1,329 @@
+ * modify it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * iotagent-json is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public
+ * License along with iotagent-json.
+ * If not, seehttp://www.gnu.org/licenses/.
+ *
+ * For those usages not covered by the GNU Affero General Public License
+ * please contact with::[contacto@tid.es]
+ *
+ * Modified by: Daniel Calvo - ATOS Research & Innovation
+ */
+
+/* jshint camelcase: false */
+
+'use strict';
+
+var iotagentMqtt = require('../../../'),
+    config = require('./config-test.js'),
+    nock = require('nock'),
+    iotAgentLib = require('iotagent-node-lib'),
+    should = require('should'),
+    async = require('async'),
+    request = require('request'),
+    utils = require('../../utils'),
+    groupCreation = {
+        url: 'http://localhost:' + config.iota.server.port + '/iot/services',
+        method: 'POST',
+        json: {
+            services: [
+                {
+                    resource: '/iot/json',
+                    apikey: 'KL223HHV8732SFL1',
+                    entity_type: 'TheLightType',
+                    trust: '8970A9078A803H3BL98PINEQRW8342HBAMS',
+                    cbHost: 'http://unexistentHost:1026',
+                    commands: [],
+                    lazy: [],
+                    attributes: [
+                        {
+                            name: 'status',
+                            type: 'Boolean'
+                        }
+                    ],
+                    static_attributes: []
+                }
+            ]
+        },
+        headers: {
+            'fiware-service': 'smartGondor',
+            'fiware-servicepath': '/gardens'
+        }
+    },
+    contextBrokerMock;
+
+describe('HTTP: Measure reception ', function() {
+    beforeEach(function(done) {
+        var provisionOptions = {
+            url: 'http://localhost:' + config.iota.server.port + '/iot/devices',
+            method: 'POST',
+            json: utils.readExampleFile('./test/deviceProvisioning/provisionDeviceHTTP.json'),
+            headers: {
+                'fiware-service': 'smartGondor',
+                'fiware-servicepath': '/gardens'
+            }
+        };
+
+        nock.cleanAll();
+
+        contextBrokerMock = nock('http://192.168.1.1:1026')
+            .matchHeader('fiware-service', 'smartGondor')
+            .matchHeader('fiware-servicepath', '/gardens')
+            .post('/v1/updateContext')
+            .reply(200, utils.readExampleFile('./test/contextResponses/multipleMeasuresSuccess.json'));
```

Comment:
This is a NGSIv1 update response... I understand this is needed to mock the entity creation upon device provision. Is that correct?

Reply:
(General comment for all the `.post('/v1/updateContext')` cases in the PR)

Reply:
However, not sure if I'm getting this correctly... in test/contextResponses/multipleMeasuresSuccess.json I see:

```
"type": "MQTTSensor",
"id": "MQTTId_2"
```

which doesn't correct to what it should be taking into account the provisioning from test/deviceProvisioning/provisionDeviceHTTP.json

```
"entity_name": "Second MQTT Device",
"entity_type": "AnMQTTDevice",
```

Maybe I'm missing something...

(This problem may affect to other similar cases along this PR)

Reply:
You're correct, this part of the tests is related to device provisioning, which as you know has not been migrated to NGSIv1.

You are again right that this is a bit strange because the mock for the context broker should return something like:
```
"type": "AnMQTTDevice",
"id": "Second MQTT Device"
```

Nevertheless, this is not very relevant since the iotagent-node-lib does not check the contents of the answer of the CB. You can check this in https://github.com/telefonicaid/iotagent-node-lib/blob/master/lib/services/devices/deviceService.js#L73. As you can see, it only checks the existence of an answer, with a body and with 200 status code.

For the NGSIv2 tests, we have just `migrated` NGSIv1 tests (i.e. https://github.com/Fiware/iot.IoTagent-JSON/blob/master/test/unit/HTTP_reveice_measures-test.js#L84). I guess that the original developers of this agent knew about this and just decided to simplify the tests, reusing always the same payload for the CB answers to device provisioning requests.

Reply:
If the response to /v1/updateContext is not relevant, I'd suggest to simplify to replace the pattern:

```
.reply(200, utils.readExampleFile('./test/contextResponses/multipleMeasuresSuccess.json'));
```

with

```
// Note /v1/updateContext response is not processed by IOTA so its content is irrelevant, as fara as it is a 200 OK
.reply(200, {});
```

This would reduce entropy (and confussions like mine :).

Reply:
This would not work because the iotagent-node-lib does check if the CB returns body... The only option to change this and to make answers coherent would be to replace this by the appropriate answers from the CB...

Reply:
`{}` is a valid response body? (valid in the sense iotagent-node-lib will not complaint about it)

Reply:
It seems to work! :+1:

Reply:
Fixed in 3dbf32d1862f7f3ca4d1658b76c6dccdeaed3874